### PR TITLE
Update smol

### PR DESCRIPTION
**Pull request by stjepang**
This PR updates `smol` to v1.0 and replaces `serde` with `json`.

Compilation times are cut roughly in half on my 8-core machine, which means a debug build now takes 27 seconds. There are still ways to go, but it seems that the largest part of compilation is related to crypto stuff (`nkeys` and `async-tls`), which will be difficult to trim down. `smol` itself takes up only around 5 seconds.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,21 +23,19 @@ panic = 'abort'
 
 [dependencies]
 async-dup = "1.2.1"
-async-tls = "0.8.0"
+async-tls = { path = "../async-tls", default-features = false, features = ["client"] }
```
Comment (derekcollison) [inline Cargo.toml]: Why using a local copy?

File: Cargo.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ panic = 'abort'
 
 [dependencies]
 async-dup = "1.2.1"
-async-tls = { path = "../async-tls", default-features = false, features = ["client"] }
+async-tls = { version = "0.10.0", default-features = false, features = ["client"] }
```
Comment (derekcollison) [inline Cargo.toml:31]: nvm ;)

**Review by derekcollison — APPROVED** [review]
In general LGTM. Love the faster compile times!

File: Cargo.toml (hunk 1 above)
Comment (stjepang) [reply Cargo.toml]: Not anymore, that was from a previous commit. I had to temporarily use my local patched version of `async-tls` until a new version of it was published today. :)

**Pull request merged**